### PR TITLE
Smartly reactivate beginend when narrowing is in effect

### PR DESCRIPTION
**Pull request by DamienCassou**
Instead of just deactivating beginend when a buffer is narrowed, this
patch widens the buffer, applies the beginend procedure, and checks if
point has moved outside the original narrowed region.

### CODE REVIEW

**Comment by DamienCassou** [comment]
Up

File: beginend.el [hunk 1]
```diff
@@ -49,18 +49,28 @@
   (re-search-backward "[^[:space:]]" nil t)
   (forward-char))
 
+(defun beginend--out-of-bounds-p (point)
+  "Return non-nil if POINT is outside [`point-min', `point-max'].
+This is possible if buffer was narrowed after POINT was stored."
+  (not (<= (point-min) point (point-max))))
```
Comment (Fuco1) [inline beginend.el:56]: This only works in 25+.  Is that our lowest-supported version?

Comment (DamienCassou) [reply beginend.el:56]: Apparently this got added in Emacs 24.4:

> Numeric comparison functions =, <, >, <=, >= can now take many arguments.

That's good enough for me.

Comment (DamienCassou) [reply beginend.el:56]: I also added tests and Travis builds this package on Emacs 24.5.

Comment (Fuco1) [reply beginend.el:56]: Oh, okey.  Somehow I was under the impression it's a 25+ feature.  Let's go ahead with this then.  But we should probably add a version constraint to the requirements.

Comment (DamienCassou) [reply beginend.el:56]: > But we should probably add a version constraint to the requirements.

beginend.el already contains:

    ;; Package-Requires: ((emacs "24.4"))

Do you have something else in mind?

**Pull request merged**